modules/cli/src/main/scala/scala/cli/internal/CliLogger.scala

```diff
@@ -83,9 +83,15 @@ class CliLogger(
         diag.setSeverity(severity.toBsp4j)
         diag.setSource("scala-cli")
 
-        for (textEdit <- textEditOpt) {
-          val bTextEdit = TextEdit(range, textEdit.newText)
-          diag.setData(bTextEdit.toJsonTree())
+        (textEditOpt, f.path) match {
+          case (Some(textEdit), Right(path)) =>
+            val bTextEdit = TextEdits.TextEdit(range, textEdit.newText)
+            val workspaceEdit = WorkspaceEdit(changes =
+              Map(path.toNIO.toUri().toString() -> Array(bTextEdit))
+            )
+            val data = DiagnosticData(edits = Array(workspaceEdit))
+            diag.setData(data.toJsonTree())
+          case _ => ()
```

Comment:
Alright, so I did what I mentioned earlier. if this ends up being a `Left` it's not a valid `URI`, so we don't process it for any `data`.